### PR TITLE
Error types for buffer mapping API

### PR DESCRIPTION
**Pull request by GabrielMajeri**
**Connections**
Part of #638 

**Description**
Adds safe error handling for buffer mapping/unmapping functions.

**Testing**
Checked with core, and ran tests with player.

### CODE REVIEW

File: wgpu-core/src/device/mod.rs [hunk 1]
```diff
@@ -2905,14 +2898,16 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
...
         device.lock_life(&mut token).map(buffer_id, ref_count);
+
+        Ok(())
     }
 
     pub fn buffer_get_mapped_range<B: GfxBackend>(
         &self,
         buffer_id: id::BufferId,
         offset: BufferAddress,
         _size: Option<BufferSize>,
```
Comment (GabrielMajeri) [inline wgpu-core/src/device/mod.rs:2909]: I just noticed this argument, what is it for? Is the intention to check that the mapped range has a valid size?

Comment (kvark) [reply wgpu-core/src/device/mod.rs:2909]: we need to check and validate it, but we aren't using it in native implementation. A mapped range is just a pointer.
The parameter came from the WebGPU API spec, where it's needed to know how much to synchronize between content and GPU processes

**Comment by kvark** [comment]
please rebase

**Comment by kvark** [comment]
bors +

**Comment by GabrielMajeri** [comment]
Rebased, should be merge-able now.

I've learned the compromise of many small PRs: easier to review, much more merge conflicts :sweat_smile:

**Pull request merged**